lib-src/rpc2/rp2gen/crout.c

```diff
@@ -1115,57 +1104,66 @@ static void pack(WHO who, VAR *parm, char *prefix, char *ptr, FILE *where)
     suffix = concat3elem("[", iterate, "]");
     switch (parm->type->type->tag) {
     case RPC2_INTEGER_TAG:
-	    fputs("    if (pack_int(buf, ", where);
+     buffer_checked = 1;
+	    fputs("    if (pack_int(bufptr, ", where);
 	    if (who == RP2_CLIENT && mode == IN_OUT_MODE) fputc('*', where);
 	    fprintf(where, "%s))\n", name);
      fputs("        goto bufferoverflow;\n", where);
 	    break;
     case RPC2_UNSIGNED_TAG:
-	    fputs("    if (pack_unsigned(buf, ", where);
+     buffer_checked = 1;
+	    fputs("    if (pack_unsigned(bufptr, ", where);
 	    if (who == RP2_CLIENT && mode == IN_OUT_MODE) fputc('*', where);
 	    fprintf(where, "%s))\n", name);
      fputs("        goto bufferoverflow;\n", where);
 	    break;
     case RPC2_BYTE_TAG:
+     buffer_checked = 1;
```

Comment:
Indentations are all over the place, sometimes they are less indented than the surrounding code, sometimes they are more. Are you using a tabstop set to something other than 8 spaces?